compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1249,8 +1329,11 @@ object SymDenotations {
     }
 
     override protected[dotc] final def info_=(tp: Type) = {
-      super.info_=(tp)
+      if (changedClassParents(infoOrCompleter, tp, completersMatter = true))
+        invalidateBaseDataCache()
+      invalidateMemberNamesCache()
```

Comment:
member names are relatively infrequently needed, so I think we lose little to invalidate them more aggressively than strictly needed. Also, names will usually be entered later using `enteredAfter`.